src/components/DeeplinkWrapper/index.website.js

```diff
@@ -0,0 +1,157 @@
+import _ from 'underscore';
+import {View} from 'react-native';
+import PropTypes from 'prop-types';
+import React, {PureComponent} from 'react';
+import deeplinkRoutes from './deeplinkRoutes';
+import FullScreenLoadingIndicator from '../FullscreenLoadingIndicator';
+import TextLink from '../TextLink';
+import * as Illustrations from '../Icon/Illustrations';
+import withLocalize, {withLocalizePropTypes} from '../withLocalize';
+import Text from '../Text';
+import styles from '../../styles/styles';
+import CONST from '../../CONST';
+import CONFIG from '../../CONFIG';
+import Icon from '../Icon';
+import * as Expensicons from '../Icon/Expensicons';
+import colors from '../../styles/colors';
+import * as Browser from '../../libs/Browser';
+
+const propTypes = {
+    /** Children to render. */
+    children: PropTypes.node.isRequired,
+
+    ...withLocalizePropTypes,
+};
+
+class DeeplinkWrapper extends PureComponent {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            appInstallationCheckStatus: this.isMacOSWeb() ? 'checking' : 'not-installed',
+        };
+    }
+
+    componentDidMount() {
+        if (!this.isMacOSWeb()) {
+            return;
+        }
+
+        let focused = true;
+
+        window.addEventListener('blur', () => {
+            focused = false;
+        });
+
+        setTimeout(() => {
+            if (!focused) {
+                this.setState({appInstallationCheckStatus: 'installed'});
+            } else {
+                this.setState({appInstallationCheckStatus: 'not-installed'});
+            }
+        }, 500);
+
+        // check if pathname matches with deeplink routes
+        const matchedRoute = _.find(deeplinkRoutes, (route) => {
+            const routeRegex = new RegExp(route.pattern);
+            return routeRegex.test(window.location.pathname);
+        });
+
+        if (matchedRoute) {
+            this.setState({deeplinkMatch: true});
+            this.openRouteInDesktopApp();
+        } else {
+            this.setState({deeplinkMatch: false});
+        }
+    }
+
+    openRouteInDesktopApp() {
+        const expensifyUrl = new URL(CONFIG.EXPENSIFY.NEW_EXPENSIFY_URL);
+        const expensifyDeeplinkUrl = `${CONST.DEEPLINK_BASE_URL}${expensifyUrl.host}${window.location.pathname}`;
+
+        // This check is necessary for Safari, otherwise, if the user
+        // does NOT have the Expensify desktop app installed, it's gonna
+        // show an error in the page saying that the address is invalid
```

Comment:
### Slack with Desktop app installed
https://user-images.githubusercontent.com/85645967/211314606-cafd71e8-ece1-425e-9c0d-04c7a9f49575.mov

### Slack without Desktop app
https://user-images.githubusercontent.com/85645967/211314646-178e7968-5470-4db7-99f3-06d30f869926.mov

Reply:
Please note that this is Safari only thing. The videos you posted are in Chrome.

Reply:
How does it work on safari for slack?

Reply:
It works the same way. Slack doesn't automatically open their webapp like we're doing. They always display the "launching" screen.

Reply:
We only show the "launching" screen if the user has the app installed, otherwise if will open the webapp automatically. There's only that Safari bug that was previously discussed.

Reply:
No worries